src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -220,11 +220,15 @@ protected EC2Cloud(String id, boolean useInstanceProfileForCredentials, String c
     public KeyPair resolveKeyPair() throws IOException {
         KeyPair keyPair = null;
 
-        if (sshKeysCredentialsId != null) {
+        if (sshKeysCredentialsId != null && !sshKeysCredentialsId.isEmpty()) {
```

Comment:
OK though better to have the corresponding setter call `fixEmpty` so the config does not store

```xml
<sshKeysCredentialsId></sshKeysCredentialsId>
```

to begin with.

Reply:
i can take a look

Reply:
it looks like the setter is allready calling `fixEmpty()` so i will remove this and test again